packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -1082,13 +1082,8 @@ class ScrollAction extends Action<ScrollIntent> {
       if (Scrollable.of(focus.context!) != null)
         return true;
       // Check for fallback scrollable with context from PrimaryScrollController
-      if (PrimaryScrollController.of(focus.context!) != null) {
-        final ScrollController? primaryScrollController = PrimaryScrollController.of(focus.context!);
-        return primaryScrollController != null
-          && primaryScrollController.hasClients
-          && primaryScrollController.position.context.notificationContext != null
-          && Scrollable.of(primaryScrollController.position.context.notificationContext!) != null;
-      }
+      final ScrollController? primaryScrollController = PrimaryScrollController.of(focus.context!);
+      return primaryScrollController != null && primaryScrollController.hasClients;
```

Comment:
I moved much of this logic to the `invoke` method. Accessing the `position` here would throw an error (the bug we are fixing), but the error did not make sense in the stack trace that would result from throwing here. 
By throwing from invoke, I thought it would make more sense.